src/app/pages/deprecation/DeprecationAction.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace app\pages\deprecation;
+
+use Slim\Psr7\Request;
+use Slim\Views\Twig;
+use app\domain\Version;
+use app\domain\util\Redirect;
+use app\domain\Artifact;
+
+class DeprecationAction
+{
+  private Twig $view;
+
+  public function __construct(Twig $view)
+  {
+    $this->view = $view;
+  }
+
+  public function __invoke(Request $request, $response, $args)
+  {
+    return $this->view->render($response, 'deprecation/deprecation.twig', [
+      'versions' => self::getVersions(),
+	  'features' => self::getFeatures()
+    ]);
+  }
+  
+  private static function getVersions(): array
+  {
+    return ["3", "4", "5", "6", "7", "8", "9.1", "9.2", "9.3"];
+  }
+  
+  private static function getFeatures(): array
+  {
+	return [
+	  new Feature(
+	    "CMS Page", 
+		"/doc/8.0/designer-guide/user-interface/web-page/html-content-in-the-cms.html",
+		"Html Dialog", 
+		"/doc/latest/designer-guide/user-interface/user-dialogs/html-dialogs.html",
+		"3",
+		"8",
+		"9.3"),		
+	  new Feature(
+	    "Macros in RTF documents", 
+		"/doc/8.0/designer-guide/user-interface/web-page/index.html",
+		"Doc Factory", 
+		"/market/doc-factory",
+		"3",
+		"8",
+		"9.3"),		
+	  new Feature(
+	    "Import Ivy 3.9 Project", 
+		"/doc/7.0/DesignerGuideHtml/ivy.processmodeling.html#ivy-project-convert39",
+		"", 
+		"",
+		"5",
+		"8",
+		"8"),		
+	  new Feature(
+	    "RIA / Rich Dialog", 
+		"/doc/7.0/DesignerGuideHtml/ivy.userinterface.html#ivy-richdialogs",
+		"Html Dialog", 
+		"/doc/latest/designer-guide/user-interface/user-dialogs/html-dialogs.html",
+		"4",
+		"7",
+		"8"),
+	  new Feature(
+	    "WebService Axis 1 & 2", 
+		"doc/8.0/designer-guide/configuration/webservice-clients.html",
+		"WebService CXF", 
+		"/doc/latest/designer-guide/configuration/webservice-clients.html",
+		"3",
+		"8",
+		"9.3"),			
+	];
+  }
+}
+
+class Feature 
+{
+  public String $name;
+  public String $nameHref;
+  public String $successor;
+  public String $successorHref;
+  public String $released;
```

Comment:
maybe a sub-class? Lifetime(released, deprected, removed)? ... so we could shorten the param list.